### PR TITLE
Remove obsolete code from basicauth example

### LINKED ISSUE #1471: OpenAPI basicauth example is outdated
The example references old code.

```
Traceback (most recent call last):
  File "app.py", line 7, in <module>
    from connexion.decorators.security import validate_scope
ModuleNotFoundError: No module named 'connexion.decorators.security'
```

### PR DESCRIPTION
**Pull request by a-a-abramov**
Fixes #1471 / #1485

Changes proposed in this pull request:

 - Remove obsolete code (see the beginning of #1485 for concrete reasoning)
 - Remove scopes completely except necessary arg in `x-basicInfoFunc` implementation

### CODE REVIEW

File: examples/openapi3/basicauth/app.py [hunk 1]
```diff
@@ -4,29 +4,16 @@
...
+    'foo': 'bar'
+}
 
-def basic_auth(username, password, required_scopes=None):
-    if username == 'admin' and password == 'secret':
-        info = {'sub': 'admin', 'scope': 'secret'}
-    elif username == 'foo' and password == 'bar':
-        info = {'sub': 'user1', 'scope': ''}
-    else:
-        # optional: raise exception for custom error response
```
Comment (Ruwann) [inline examples/openapi3/basicauth/app.py:17]: nit but can we leave this comment in here to indicate that this is also possible to do from a security function?

File: examples/openapi3/basicauth/app.py [hunk 2]
```diff
@@ -4,29 +4,16 @@
...
-    if required_scopes is not None and not validate_scope(required_scopes, info['scope']):
-        raise OAuthScopeProblem(
-                description='Provided user doesn\'t have the required access rights',
-                required_scopes=required_scopes,
-                token_scopes=info['scope']
-            )
-
-    return info
-
+def basic_auth(username, password, required_scopes=None) -> dict:
```
Comment (Ruwann) [inline examples/openapi3/basicauth/app.py:13]: You don't have to typehint the return type here, but if you want to, it should be `typing.Optional[dict]` as it's possible to return `None`

We also don't need the `required_scopes` argument here anymore as well (the example should still work if you leave it out)

Comment (Ruwann) [inline examples/openapi3/basicauth/app.py:13]: You don't have to typehint the return type here, but if you want to, it should be `typing.Optional[dict]` as it's possible to return `None`

File: examples/openapi3/basicauth/app.py [hunk 3]
```diff
@@ -4,29 +4,16 @@
 '''
 
 import connexion
-from connexion.decorators.security import validate_scope
-from connexion.exceptions import OAuthScopeProblem
 
+PASSWD={
```
Comment (Ruwann) [inline examples/openapi3/basicauth/app.py:8]:
```suggestion
PASSWD = {
```

Comment (Ruwann) [inline examples/openapi3/basicauth/app.py:8]:
```suggestion
PASSWD = {
```

**Review by Ruwann — CHANGES_REQUESTED** [review]
Thanks for the PR! Left some small comments, but should be good to merge when addressed :)

There's also still a reference in the swagger2 basicauth example to the usage of scopes here:
https://github.com/spec-first/connexion/blame/main/examples/swagger2/basicauth/README.rst#L16-L19

We can just delete these lines from the README imo

File: examples/openapi3/basicauth/app.py (hunk 2 above)
Comment (a-a-abramov) [reply examples/openapi3/basicauth/app.py:13]: Ah, good catch. Totally missed it!

**Comment by a-a-abramov** [comment]
I sort of made 1 more logical step and made app.py-s and README.rst-s almost identical for both swagger and openapi3. Code duplication bothered me a bit, but they are examples after all.

**Review by Ruwann — APPROVED** [review]
Awesome, thanks a lot!

I agree that the code duplication is not that bad for examples, it's better to have each example complete on its own.

**Comment by a-a-abramov** [comment]
Cool, thanks a lot!

**Pull request merged**